### DISCUSSION TITLE
Inconsistent default parameters

### DISCUSSION THREAD

**Discussion by juseg** [pism/pism#487] [Q&A]
Extract from the current default config in dev:

```
pism_config:beta_CC = 7.9e-8;
pism_config:beta_CC_doc = "K Pa-1; Clausius-Clapeyron constant [@ref Luethi2002]";
pism_config:water_melting_point_temperature = 273.15;
pism_config:water_melting_point_temperature_doc = "K; melting point of pure water";
pism_config:water_latent_heat_fusion = 3.34e5;
pism_config:water_latent_heat_fusion_doc = "J kg-1; latent heat of fusion for water [@ref AschwandenBlatter]";
pism_config:ice_density = 910.0;
pism_config:ice_density_doc = "kg m-3; = rho_i; density of ice in ice sheet";
```

According to the Clapeyron equation,

```
Delta T / Delta P = T * Delta v / L
```

we have:

```
beta = Delta T / Delta P
     = T * Delta v / L
     = T / L * (1/v_i - 1/v_w)
     = T / L * (1/rho_i - 1/rho_w)
```

But using the above values for `T`, `L` and `rho_i`, and `rho_w = 1000`, we get `beta = 8.088e-8 K Pa-1`. I came accross this inconsistence when writing my thesis and wonder if, although small, it may cause energy "leaks" in the model.

If this is the case, perhaps the most rigorous fix would be to remove one of these parameters from the configuration file and instead compute it from the three other ones. Arguably that would be the "Clapeyron constant" `beta`. On the other hand, I see the rationale behind using the experimental value `beta = 7.9e-8` from Lüthi et al. (2002). Maybe the default for ice density should be changed? I am not sure what would be the best fix here...

**Comment by aaschwanden** [comment]
Julien,

we would have to dig deep into the continuum equations, off the top of my head I fail to see how this could produce an energy leak (but yes, my intuition could be wrong.

```
  = T / L * (1/v_i - 1/v_w) 
```

is correct if you assume that the water is free of any air bubble. The value from Luethi et al. (2002) is indicative of the presence of air, so you probably would have to write

```
= T / L * (1/v_i - 1/v_w - 1/v_air)
```

with the air volume virtually unknown. From our current understanding of temperate ice, I don't think we are able to calculate beta from first principles, as you suggest. I see is as a tuning parameter. But again, I could be wrong.

**Comment by bueler** [comment]
Julien,

I agree with Andy that I don't think _this_ issue is a energy-conservation leak.  (There are other leaks I am more worried about, like at ice sheet margins.)

In PISM, and in Aschwanden et al (2012), we are defining internal energy a certain way, and in particular we assume the ice mixture is incompressible.  This means our approximation of conservation is less-than-physical, but we can still conserve this defined energy.

Yes, pressure does affect pressure-melting temperature.  Consider a chunk of ice/water mixture which is experiencing no conductive losses (i.e. take the empirical conductivity to be zero).  Suppose it is advected from one pressure level (depth) to a different one.  Then the enthalpy does change according to the pressure-melting temperature (and thus `beta` determines the change).  But this is the only change to the enthalpy of our incompressible chunk.  If we move the chunk back to the old pressure level we are back to the same enthalpy.  Along the way we did no work (because no compression), so the fact that enthalpy is back to the old value is correct.

Note that a better enthalpy-based conservation of energy model would be compressible.  In that compressible theory we would want the Clapeyron equation to hold, and then we would have to decide how air bubbles would shift the value.  But we don't have that now.

By the way, I am not very confident in my understanding of conservation of energy!  I would love to see a critical re-analysis of the theory in Aschwanden et al (2012).  You?
